source/common/upstream/load_balancer_impl.h

```diff
@@ -555,16 +576,16 @@ class RoundRobinLoadBalancer : public EdfLoadBalancerBase {
     initialize();
   }
 
-  RoundRobinLoadBalancer(
-      const PrioritySet& priority_set, const PrioritySet* local_priority_set, ClusterLbStats& stats,
-      Runtime::Loader& runtime, Random::RandomGenerator& random, uint32_t healthy_panic_threshold,
-      const envoy::extensions::load_balancing_policies::round_robin::v3::RoundRobin&
-          round_robin_config,
-      TimeSource& time_source)
+  RoundRobinLoadBalancer(const PrioritySet& priority_set, const PrioritySet* local_priority_set,
+                         ClusterLbStats& stats, Runtime::Loader& runtime,
+                         Random::RandomGenerator& random, uint32_t healthy_panic_threshold,
+                         OptRef<const TypedRoundRobinLbConfig> round_robin_config,
+                         TimeSource& time_source)
       : EdfLoadBalancerBase(
             priority_set, local_priority_set, stats, runtime, random, healthy_panic_threshold,
-            LoadBalancerConfigHelper::localityLbConfigFromProto(round_robin_config),
-            LoadBalancerConfigHelper::slowStartConfigFromProto(round_robin_config), time_source) {
+            round_robin_config.has_value() ? round_robin_config->localityLbConfig() : absl::nullopt,
+            round_robin_config.has_value() ? round_robin_config->slowStartConfig() : absl::nullopt,
```

Comment:
Can we change the EdfLoadBalancerBase's parameter type to `const absl::optional<>&`, then we needn't this copy anymore.

Reply:
Of course. But this PR is suspended temporarily. So, you can also delay this review. Thanks.

Reply:
Will defer this optimization to new PR or land it after the legacy code path is removed.
This PR is too big now, I don't want to add new complexity. orz.